packages/machine/test/integration/test-response-sink.ts

```diff
@@ -157,16 +168,14 @@ export class TestResponseSink implements cf.node.ResponseSink {
     message: InternalMessage,
     next: Function,
     context: Context
-  ): Promise<cf.utils.Signature> {
+  ): Promise<ethers.utils.Signature> {
     const operation: CfOperation = getFirstResult(
       Instruction.OP_GENERATE,
       context.results
     ).value;
     const digest = operation.hashToSign();
-    const sig = this.signingKey.signDigest(digest);
-    return Promise.resolve(
-      new cf.utils.Signature(sig.recoveryParam! + 27, sig.r, sig.s)
-    );
+    const { recoveryParam, r, s } = this.signingKey.signDigest(digest);
+    return Promise.resolve({ r, s, v: recoveryParam! + 27 });
```

Comment:
Does this need to be wrapped in a Promis.resolve? Also the return type is not an `ethers.utils.Signature` anymore

Reply:
Yeah, `signMyUpdate` is expected to return a Promise.

Also, if that were the case (return type =/= `ethers.utils.Signature`) then the Typescript build would fail, which it doesn't. ethers.utils.Signature is just any object with an `r`,`s` and `v`